synapse/handlers/federation_event.py

```diff
@@ -897,6 +899,9 @@ async def _process_pulled_event(
                     backfilled=backfilled,
                 )
         except FederationError as e:
+            if backfilled:
+                await self._store.record_event_backfill_attempt(event_id)
```

Comment:
should we clear the failed backfill attempts when we successfully pull a backfilled event?

Reply:
We do clear it alongside when we also clear it as a backward extremity, see the `DELETE FROM event_backfill_attempts` part of this

Reply:
Added a comment to make this more clear 👍

> Clear out the failed backfill attempts after we successfully pulled the event.